Setup CircleCI to store build artifacts

nuilewis opened the pull request:
This adds a config to deploy build artifacts to GitHub Releases which can be used for the Tables tests.

nuilewis committed:
Added config to deploy snapshot builds to Github Releases

Redeem-Grimm-Satoshi commented:
@nuilewis Awesome, currently having one failed test?

wbrunette commented:
We changed the way we are publishing. See Androidlibrary and AndroidCommon
https://github.com/odk-x/androidlibrary/blob/upgrade-jdk11/.circleci/config.yml

nuilewis commented:
> We changed the way we are publishing. See Androidlibrary and AndroidCommon https://github.com/odk-x/androidlibrary/blob/upgrade-jdk11/.circleci/config.yml

Uh, okay. I figured this is just for the android library and android common since they are published as packages.

I added a confIg to push the builds apks from the CircleCI build jobs to GitHub, so we can use them for the Tables tests. I remember we spoke one time, and we discussed how the current setup for tables was too complicated because it had to make multiple requests to the circleci api to get the build artifacts from previous jobs then download the apks before installing for the tests, then you suggested we publish the build artifacts to Github rather.

nuilewis commented:
> @nuilewis Awesome, currently having one failed test?

The tests are failing because they are configured to run on the xlarge resource class on circleci which my personal account doesn't have access to.

wbrunette commented:
> > We changed the way we are publishing. See Androidlibrary and AndroidCommon https://github.com/odk-x/androidlibrary/blob/upgrade-jdk11/.circleci/config.yml
> 
> Uh, okay. I figured this is just for the android library and android common since they are published as packages.
> 
> I added a confIg to push the builds apks from the CircleCI build jobs to GitHub, so we can use them for the Tables tests. I remember we spoke one time, and we discussed how the current setup for tables was too complicated because it had to make multiple requests to the circleci api to get the build artifacts from previous jobs then download the apks before installing for the tests, then you suggested we publish the build artifacts to Github rather.

Your approach is valid; however, a "snapshot" build should not be an official release. It should be like the libraries where we test things BEFORE they become an "official release". We are using the packages to store the "snapshots".

nuilewis commented:
> > > We changed the way we are publishing. See Androidlibrary and AndroidCommon https://github.com/odk-x/androidlibrary/blob/upgrade-jdk11/.circleci/config.yml
> > 
> > 
> > Uh, okay. I figured this is just for the android library and android common since they are published as packages.
> > I added a confIg to push the builds apks from the CircleCI build jobs to GitHub, so we can use them for the Tables tests. I remember we spoke one time, and we discussed how the current setup for tables was too complicated because it had to make multiple requests to the circleci api to get the build artifacts from previous jobs then download the apks before installing for the tests, then you suggested we publish the build artifacts to Github rather.
> 
> Your approach is valid; however, a "snapshot" build should not be an official release. It should be like the libraries where we test things BEFORE they become an "official release". We are using the packages to store the "snapshots".

Okay, I understand.
Now I am a bit confused, if we are using packages for the build apks of the apps, does that mean we would have to consume them in the build.gradle like an actual package? like the way we are currently doing with androidlibray and androidcommon? That's not really what we want right?

Or is it possible to download the "package" files as just files we can then use anywhere?

wbrunette commented:
you are correct, it's not supported yet. 

What are your thoughts on using GitHub actions and artifacts? https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

wbrunette commented:
Maybe just use the artifact system in CircleCI instead? https://circleci.com/docs/artifacts/#downloading-all-artifacts-for-a-build-on-circleci

nuilewis commented:
> you are correct, it's not supported yet.
> 
> What are your thoughts on using GitHub actions and artifacts? https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

Uhm, I think adding a layer of GitHub actions might further complicate the build system, because CircleCI already has its own artifact system, we might as well just use that in my opinion

nuilewis commented:
> Maybe just use the artifact system in CircleCI instead? https://circleci.com/docs/artifacts/#downloading-all-artifacts-for-a-build-on-circleci

Alright then, yeah this was similar to the initial approach I did, storing the build files as artifacts then downloading them when running the tables tests.

nuilewis committed:
Setup CircleCI for storing build artifacts

nuilewis commented:
Hello @wbrunette
I have updated the config to store the build artifacts in CircleCI

wbrunette commented on .circleci/config.yml:
```diff
@@ -55,13 +55,49 @@ jobs:
+            # - services_app/build/outputs/apk/snapshot/debug/services_app-snapshot-debug.apk
+
+
+#  deploy:
```
Please remove commented out code.

nuilewis replied:
Alright, done

nuilewis committed:
Removed comments
